### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
I'm open to adding a changelog item, but as I generally get pushback on some changes in long sets, I'm disinclined to write it proactively as I don't know what it'll say.

Fwiw, here's the tooling output that reported the items that I'm fixing: https://github.com/jsoref/pytest/actions/runs/8286348273#summary-22676128090
And here's a report showing those items have been addressed: https://github.com/jsoref/pytest/actions/runs/8286348842#summary-22676129381

### CODE REVIEW

File: doc/en/announce/release-2.0.0.rst [hunk 1]
```diff
@@ -62,7 +62,7 @@ New Features
 - new "-q" option which decreases verbosity and prints a more
   nose/unittest-style "dot" output.
 
-- many many more detailed improvements details
+- many, many, more detailed improvements details
```
Comment (jsoref) [inline doc/en/announce/release-2.0.0.rst:65]: generally in English, doubled words aren't appropriate without punctuation, and often they're just errant.

File: doc/en/announce/release-2.4.0.rst [hunk 2]
```diff
@@ -181,7 +181,7 @@ Bug fixes:
   partially failed (finalizers would not always be called before)
 
 - fix issue320 - fix class scope for fixtures when mixed with
-  module-level functions.  Thanks Anatloy Bubenkoff.
+  module-level functions.  Thanks Anatoly Bubenkoff.
```
Comment (jsoref) [inline doc/en/announce/release-2.4.0.rst:184]: This matches other instances of what I have to presume is the same entity.

File: doc/en/announce/release-2.5.0.rst [hunk 3]
```diff
@@ -83,7 +83,7 @@ holger krekel
   Thanks Ralph Schmitt for the precise failure example.
 
 - fix issue244 by implementing special index for parameters to only use
-  indices for paramentrized test ids
+  indices for parametrized test ids
```
Comment (jsoref) [inline doc/en/announce/release-2.5.0.rst:86]: there's an amusing thing elsewhere about a serious of variations of this word, but this isn't one of them...

File: doc/en/announce/release-2.6.0.rst [hunk 4]
```diff
@@ -73,7 +73,7 @@ holger krekel
 - cleanup setup.py a bit and specify supported versions. Thanks Jurko
   Gospodnetic for the PR.
 
-- change XPASS colour to yellow rather then red when tests are run
+- change XPASS colour to yellow rather than red when tests are run
```
Comment (jsoref) [inline doc/en/announce/release-2.6.0.rst:76]: technically this is grammatical, but most spell checkers have been able to flag rudimentary grammar errors, so there are a handful of fixes for things like this...

File: doc/en/changelog.rst [hunk 5]
```diff
@@ -1853,7 +1853,7 @@ Bug Fixes
   the ``tmp_path``/``tmpdir`` fixture). Now the directories are created with
   private permissions.
 
-  pytest used to silently use a pre-existing ``/tmp/pytest-of-<username>`` directory,
+  pytest used to silently use a preexisting ``/tmp/pytest-of-<username>`` directory,
```
Comment (jsoref) [inline doc/en/changelog.rst:1856]: both flavors were preexisting in this repository :)

File: doc/en/changelog.rst
Comment (jsoref) [inline doc/en/changelog.rst:1]: Some projects don't like changing changelogs. Happy to drop as desired.

File: doc/en/deprecations.rst [hunk 6]
```diff
@@ -427,7 +427,7 @@ Now :class:`~pytest.Class` collects the test methods directly.
 Most plugins which reference ``Instance`` do so in order to ignore or skip it,
 using a check such as ``if isinstance(node, Instance): return``.
 Such plugins should simply remove consideration of ``Instance`` on pytest>=7.
-However, to keep such uses working, a dummy type has been instanted in ``pytest.Instance`` and ``_pytest.python.Instance``,
+However, to keep such uses working, a dummy type has been instanced in ``pytest.Instance`` and ``_pytest.python.Instance``,
```
Comment (jsoref) [inline doc/en/deprecations.rst:430]: I couldn't be certain that this shouldn't be `instantiated`

File: doc/en/naming20.rst [hunk 7]
```diff
@@ -8,7 +8,7 @@ If you used older version of the ``py`` distribution (which
 included the py.test command line tool and Python name space)
 you accessed helpers and possibly collection classes through
 the ``py.test`` Python namespaces.  The new ``pytest``
-Python module flaty provides the same objects, following
+Python module flatly provides the same objects, following
```
Comment (jsoref) [inline doc/en/naming20.rst:11]: It could be that there's a module `flaty`, but...

File: doc/en/reference/plugin_list.rst [hunk 8]
```diff
@@ -47,7 +47,7 @@ This list contains 1411 plugins.
...
-   :pypi:`pytest-agent`                             Service that exposes a REST API that can be used to interract remotely with Pytest. It is shipped with a dashboard that enables running tests in a more convenient way.                                                                                                                                                                                                                 Nov 25, 2021    N/A                    N/A
+   :pypi:`pytest-agent`                             Service that exposes a REST API that can be used to interact remotely with Pytest. It is shipped with a dashboard that enables running tests in a more convenient way.                                                                                                                                                                                                                 Nov 25, 2021    N/A                    N/A
```
Comment (jsoref) [inline doc/en/reference/plugin_list.rst:50]: I couldn't tell if this is a list of descriptions that come from the modules, in which case the fixes would belong in the modules -- but searches didn't seem to show lots of indications where I'd expect if that was the case, so...

File: src/_pytest/legacypath.py [hunk 9]
```diff
@@ -384,7 +384,7 @@ def Config_inifile(self: Config) -> Optional[LEGACY_PATH]:
     return legacy_path(str(self.inipath)) if self.inipath else None
 
 
-def Session_stardir(self: Session) -> LEGACY_PATH:
+def Session_startdir(self: Session) -> LEGACY_PATH:
```
Comment (jsoref) [inline src/_pytest/legacypath.py:387]: It's possible this is a public export -- I can't tell.

File: src/_pytest/stepwise.py [hunk 10]
```diff
@@ -40,7 +40,7 @@ def pytest_addoption(parser: Parser) -> None:
 @pytest.hookimpl
 def pytest_configure(config: Config) -> None:
     if config.option.stepwise_skip:
-        # allow --stepwise-skip to work on it's own merits.
+        # allow --stepwise-skip to work on its own merits.
```
Comment (jsoref) [inline src/_pytest/stepwise.py:43]: the previous text would mean `it is` which seems very wrong

File: testing/test_runner.py [hunk 11]
```diff
@@ -408,7 +408,7 @@ def test_func():
         # assert rep.outcome.when == "setup"
         # assert rep.outcome.where.lineno == 3
         # assert rep.outcome.where.path.basename == "test_func.py"
-        # assert instanace(rep.failed.failurerepr, PythonFailureRepr)
+        # assert isinstance(rep.failed.failurerepr, PythonFailureRepr)
```
Comment (jsoref) [inline testing/test_runner.py:411]: It's possible this was `instance`, but I didn't think that was a thing

**Review by nicoddemus — APPROVED** [review]
Looks great, thanks!

Usually we would squash the commits before merging for a mostly automated PR like this one, but perhaps it makes sense to keep those commits because they do add context to each change?

**Comment by jsoref** [comment]
@nicoddemus: are you waiting for something from me?

**Comment by nicoddemus** [comment]
@jsoref no, thanks for the ping. I was waiting to see if others would like to review too, but merging now.

**Pull request merged**